Домашнее задание к лекции №5

arsvrn opened the pull request.

arsvrn committed:
Домашнее задание к лекции №5

SHvatov commented on src/main/java/com/tbank/edu/hw5/repository/CategoryRepository.java:
```diff
@@ -0,0 +1,35 @@
+import java.util.concurrent.ConcurrentHashMap;
+
+@Repository
+public class CategoryRepository implements ICategoryRepository {
```
`ICategoryRepository` - `I<interface name>` - это больше про C# и компанию. Да и в скором времени вы переедете на Spring Data, и данный класс станет интерфейсом. Давай построим вот такую иерархию:

```
              CrudRepository<T>
           |                    |
CategoryRepository    LocationRepository
```

```java
AbstractCrudRepository implements CrudRepository
CategoryRepositoryImpl extends AbstractCrudRepository implements CategoryRepository
Location...
```

SHvatov commented on src/main/java/com/tbank/edu/hw5/service/ExternalApiService.java:
```diff
@@ -0,0 +1,25 @@
+    private final RestTemplate restTemplate = new RestTemplate();
+
+    public List<Category> fetchCategories() {
+        Category[] categories = restTemplate.getForObject("https://kudago.com/public-api/v1.4/place-categories", Category[].class);
```
1. Урлы в константы
2. Стоит добавить обработку ошибок

SHvatov commented on src/main/java/com/tbank/edu/hw5/service/ExternalApiService.java:
```diff
@@ -0,0 +1,25 @@
+import java.util.List;
+
+@Service
+public class ExternalApiService {
```
Не вижу его использований и пункта `Кастомное поведение при старте`

SHvatov reviewed (changes requested):
@arsvrn оставил комментарии, посмотри их, пожалуйста.

arsvrn committed:
Доработка ДЗ 5

1. Изменена иерархия репозитория
2. Исправлен ExternalApiService
3. Добавлено кастомное поведение при старте

arsvrn commented:
@SHvatov доработал код, посмотри, пожалуйста

SHvatov reviewed (approved):
@arsvrn 👍
